src/protocol/aggregation/mod.rs

```diff
@@ -144,3 +159,44 @@ where
         })
         .collect::<Vec<_>>()
 }
+
+#[cfg(all(test, unit_test))]
+mod tests {
+    use super::aggregate_values;
+    use crate::{
+        ff::{Fp32BitPrime, Gf2},
+        protocol::context::{UpgradableContext, Validator},
+        secret_sharing::BitDecomposed,
+        test_fixture::{Reconstruct, Runner, TestWorld},
+    };
+
+    #[tokio::test]
+    pub async fn aggregate() {
+        const CONTRIBUTION_BITS: u32 = 8;
+        const EXPECTED: u128 = 36;
+
+        const INPUT: &[u32] = &[0, 0, 18, 0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 10, 0, 0, 6, 0];
+
+        let world = TestWorld::default();
+        let result = world
+            .semi_honest(
+                INPUT.iter().map(|&value| {
+                    BitDecomposed::decompose(CONTRIBUTION_BITS, |i| {
+                        Gf2::try_from((u128::from(value) >> i) & 1).unwrap()
+                    })
+                }),
```

Comment:
Didn't we normally hide this type of test logic within a trait that performs this logic across all tests?